File: python-algorithm/ds_9_bit_operation/L268.py
# -*- coding: utf-8 -*-


# 利用值和下标的异或运算
# 0-n每个值和下标异或，缺少的下标是n
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = 0
        for i in range(len(nums)):
            res ^= nums[i]
            res ^= i
        return res^len(nums)

File: ds_7_array_inplace/L268.cpp

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        long long sum = (n+1)*n/2;
        for(int i=0; i<nums.size(); i++)
            sum -= nums[i];
        return sum;
    }
};

//没出现在序列中的那个数就是 (和总和的差距)


//也可以用位运算解决